src/Configuration.php

```diff
@@ -212,21 +212,47 @@ public function getAutoCommit(): bool
     }
 
     /**
-     * @param Middleware[] $middlewares
+     * @deprecated setMiddlewares is deprecated since doctrine/dbal 3.6. Use setMiddleware instead.
+     *
+     * @param Middleware[] $middleware
+     *
+     * @return $this
+     */
+    public function setMiddlewares(array $middleware): self
+    {
+        trigger_error('setMiddlewares is deprecated since doctrine/dbal 3.6. Use setMiddleware instead.', E_USER_DEPRECATED);
```

Comment:
We use the doctrine/deprecations library in this repository instead of calling `trigger_error()` directly.

Reply:
Indeed, I didn't pay attention to that. Thank you.